services/BasicTypes.proto

```diff
@@ -99,6 +99,53 @@ message TokenTransferList {
     repeated NftTransfer nftTransfers = 3; // Applicable to tokens of type NON_FUNGIBLE_UNIQUE. Multiple list of NftTransfers, each of which has a sender and receiver account, including the serial number of the NFT
 }
 
+/* Representing the percentage of tokens to be transfers according to the TokenUnitFee*/
+message Fraction {
+    uint64 numerator = 1; // numerator of the fraction
+    uint64 denominator = 2; // denominator of the fraction
+}
+
+/* Fraction of the Token to be collected along with its floor if the transfer amount is low*/
+message FractionalFee {
+    Fraction fractionOfValueToCollect = 1;
+    uint64 minimumValueToCollect = 2;
+}
+
+/* A list of accounts and amounts that are involved in a Token's HbarFee*/
+message CustomHbarFees {
+    TransferList customHbarFees = 1;
+}
+
+/* A list of tokenIds and amounts representing the transferred out (negative) or into (positive) amounts according to a Token's TokenUnitFee*/
+message CustomTokenUnitFees {
```

Comment:
Is this message just for fees or for multiple token transfers?

Reply:
This is for adding the + / - amounts in token units which resulted because of custom fees, in the transaction records.